.github/workflows/format-and-lint.yaml

```diff
@@ -0,0 +1,34 @@
+on:
+  workflow_call:
+
+jobs:
+  format-and-lint:
+    runs-on: ubuntu-24.04
+    steps:
+      - uses: actions/checkout@v5
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: 3.12
+      # Change ruffConfig to main before merge
+      - name: Install dependencies
+        run: |
+          wget https://raw.githubusercontent.com/mdolab/.github/ruffConfig/.pre-commit-config.yaml
+          pip install pre-commit
+          # If there's already a ruff.toml file in the repo, we should rename it to ruff.toml-project, then download the mdolab ruff.toml file edit it to put the line `extend = "ruff.toml-project"` at the top so that ruff will use both configurations.
+          if [[ -f "ruff.toml" ]]; then
+            mv ruff.toml ruff.toml-project;
+          fi
+          wget https://raw.githubusercontent.com/mdolab/.github/ruffConfig/ruff.toml
+          if [[ -f "ruff.toml-project" ]]; then
+              sed -i '1s/^/extend = "ruff.toml-project" \n/' ruff.toml
+          fi
+          echo "Ruff config:"
```

Comment:
Related to the above comment, for users that are developing locally and want to run the same thing, its seems like they would need to manually add `extend = <path-to-global-ruff.toml>` to all project files?  We could add/have a script that does this automatically that could be used both here and locally.

Reply:
Its not quite as difficult with `flake8` since on the CLI you can just add the `--append-config` flag with the extra config file. No file editing needed. This does not seem to exist in `ruff`.